lua/lspconfig/server_configurations/haxe_language_server.lua

```diff
@@ -1,18 +1,36 @@
 local util = require 'lspconfig.util'
 
+local function find_hxml(path)
+  return vim.fs.find(function(name)
+    return name:match '.hxml$'
+  end, { path = path, type = 'file' })
+end
+
 return {
   default_config = {
     cmd = { 'haxe-language-server' },
     filetypes = { 'haxe' },
-    root_dir = util.root_pattern '*.hxml',
+    root_dir = util.root_pattern('*.hxml', '.git'),
     settings = {
       haxe = {
         executable = 'haxe',
       },
     },
-    init_options = {
-      displayArguments = { 'build.hxml' },
-    },
+    init_options = {},
+    on_new_config = function(new_config, new_root_dir)
+      if new_config.init_options.displayArguments then
+        print 'displayArguments already set. Skipping'
```

Comment:
use vim.notify replace print

Reply:
This one I left in by mistake while debugging. I've replaced the other one with `vim.notify`.